### ISSUE TITLE
Question : CRUD with functional approach DTO validation not working

### ISSUE THREAD

**Issue by priyashpatil** [nestjs/nest#4736]
Trying to achieve what @kamilmysliwiec  is doing in [Kamil's event video](https://youtu.be/jo-1EUxMmxc?list=PLcX0SGlW67YTvAMgMD4lvzCfc6NKaZQph&t=2521).
I can work with services using `@Inject`. But stuck at how do I pass Dtos or Classes. I'm new to typescript & node. I have tried generics.

## Validation not working
[git repo for reproduction](https://github.com/priyashpatil/crud-test)

```
import { Controller, Get, Type, Post, Body, UsePipes, ValidationPipe } from '@nestjs/common';
import { IsNotEmpty } from 'class-validator';

export interface ICrudController<T> {
  find(): Promise<T>;
  create(dto: T): Promise<T>;
}

export function CrudController<T>(prefix: string): Type<ICrudController<any>> {
  @Controller(prefix)
  class CrudControllerHost {
    @Get()
    async find() {
      return `Get all crud`;
    }

    @Post()
    @UsePipes(new ValidationPipe())
    async create(@Body() dto: T) {
      return dto;
    }
  }
  return CrudControllerHost;
}

// Using crud controller

// dto
export class CreatePostDto {
  @IsNotEmpty()
  title: string;
}

// extenting crudcontroller
export class PostsController extends CrudController<CreatePostDto>(
  'posts',
) {}

```

**Comment by kamilmysliwiec** [comment]
This is something that I've tried to explain in this lecture: https://www.youtube.com/watch?v=896laTifoMw

TL;DR - It's impossible to reflect generics

You can try looking into the source of this library https://github.com/nestjsx/crud - it shows what you have to do in order to enable validation.

Please, use our [Discord](https://discord.gg/G7Qnnhy) channel (support) for such questions. We are using GitHub to track bugs, feature requests, and potential improvements.

**Comment by benamar** [comment]
I went throw a similar problem, solved with
`@UsePipes(new ValidationPipe())
@UseInterceptors(ClassSerializerInterceptor)
@Controller('users')
@Crud({
  model: {
    type: UserEntity
  },
  dto: {
    create: CreateUserDTO,
    update: CreateUserDTO,
  },`
